### ISSUE TITLE
Serve engine docs (doxygen) on macaulay2.com

### ISSUE THREAD

**Issue by mahrud** [Macaulay2/M2#2586]
The gh-pages branch added after #2553 has added more than 200MB to the size of the repository, and every week nearly every file in the branch is updated, so I think it should be removed from this repository.

Having the engine documentation online sounds useful, but there's no need for tracking the changes across thousands of auto-generated html files via git, so I think it should be moved out of git and onto the server hosting macaulay2.com.

cc: @d-torrance

**Comment by d-torrance** [comment]
Yikes -- I didn't realize how big that branch was!  I've opened #2588 to remove the workflow.  Someone with the proper permissions will need to delete the branch itself.

**Comment by mahrud** [comment]
It still remains to:
- [ ] remove the existing `gh-pages` branch
- [ ] generate and serve the doxygen documentation on the macaulay2.com server

**Comment by d-torrance** [comment]
FYI, `gh-pages` is back since the workflow was run again on Sunday.  This should stop once #2588 is merged into `master`.

**Comment by mahrud** [comment]
Can we still have this set up somewhere?

**Comment by d-torrance** [comment]
One option would be to create a new repository under the Macaulay2 organization that consists of a single workflow that clones the main Macaulay2 repository every week or month or whatever, generates the engine documentation, and publishes it to GitHub pages.  Then it wouldn't muddy up the main repository.

**Comment by d-torrance** [comment]
Here's a draft: https://github.com/d-torrance/M2-engine-docs

**Comment by mahrud** [comment]
My preference would be to not use git at all, especially given that doxygen is not git friendly (updates every single file every time) and that Dan already has a server with storage and nightly tests that this can easily be incorporated into. But I'm not the one doing the work so it's not up to me.

**Comment by d-torrance** [comment]
> **Written by Claude** (Claude Opus 5, via Claude Code), not by @d-torrance, whose account posted it -- please weigh it accordingly.

Both halves of this are in place, so closing.

The branch is gone — `git ls-remote origin refs/heads/gh-pages` comes back empty, so the 200 MB of
weekly-rewritten HTML is no longer in this repository. That was the actual complaint here.

And the engine documentation is online and maintained: the draft linked above,
[`d-torrance/M2-engine-docs`](https://github.com/d-torrance/M2-engine-docs), publishes to
<https://d-torrance.github.io/M2-engine-docs/>, which is live and was last built three days ago.

Being straight about what that is and isn't: it is GitHub Pages under a personal account, not
macaulay2.com as the title asks, and it uses git — which is what @mahrud said he would rather avoid,
preferring Dan's existing server. That preference was not followed, and it is worth reopening or
filing fresh if hosting it on macaulay2.com still matters. But the repository is no longer carrying
the weight, and the docs are reachable, which is what this was for.

<sub>Triaged in a sweep of the open issues; source of truth: [`bug-triage/issues.tsv`](https://github.com/d-torrance/M2/blob/bug-triage/bug-triage/issues.tsv).</sub>
